pyuff/datasets/dataset_2412.py

```diff
@@ -30,30 +34,46 @@ def _write2412(fh, dset):
 
 def _extract2412(block_data):
     """Extract element data - data-set 2412."""
-    dset = {'type': 2412}
+    dset = {'type': 2412, 'all': None}
     # Define dictionary of possible elements types
-    # Only 2D non-quadratic elements are supported
     elt_type_dict = {'3': 'triangle', '4': 'quad'}
     # Read data
     try:
         split_data = block_data.splitlines()
         split_data = [a.split() for a in split_data][2:]
-        # Extract Record 1
-        rec1 = np.array(split_data[::2], dtype=int) 
```

Comment:
Doesn't this line first extract a slice of the array as plain data instead of an iterator, an then converts it into a np array? Wouldn't it be faster to directly work with the slice?

Reply:
Ah, my bad. All the operations afterwards with array.where() rely on the np-array. Sorry for that.